plugins/processors/scale/scale.go

```diff
@@ -0,0 +1,116 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package Scale
+
+import (
+	_ "embed"
+	"fmt"
+
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/processors"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (*Scale) SampleConfig() string {
+	return sampleConfig
+}
+
+type Scaling struct {
+	InMin  float64  `toml:"input_minimum"`
+	InMax  float64  `toml:"input_maximum"`
+	OutMin float64  `toml:"output_minimum"`
+	OutMax float64  `toml:"output_maximum"`
+	Fields []string `toml:"fields"`
+
+	factor      float64
+	fieldFilter filter.Filter
+}
+
+type Scale struct {
+	Scalings []Scaling       `toml:"scaling"`
+	Log      telegraf.Logger `toml:"-"`
+}
+
+func (s *Scaling) Init() error {
+	scalingFilter, err := filter.Compile(s.Fields)
+
+	if err != nil {
+		return fmt.Errorf("could not compile filter: %w", err)
+	}
+
+	s.fieldFilter = scalingFilter
+
+	if s.InMax == s.InMin {
+		return fmt.Errorf("minumum and maximum are equal for fields %s", strings.Join(s.Fields, ","))
+	}
```

Comment:
Shouldn't this also be checked for the Out fields? I know it won't give a divide by 0 error, but the factor and thus the results will always be 0.

Also, a typo:
```suggestion
	if s.InMax == s.InMin {
		return fmt.Errorf("minimum and maximum are equal for fields %s", strings.Join(s.Fields, ","))
	}
```